### DISCUSSION TITLE
SCtransform: Will not return corrected UMI because residual type is not set to 'pearson'

### DISCUSSION THREAD

**Discussion by t1nnenissen** [satijalab/seurat#6850] [Q&A]
Hi there, 

I am running SCtransform with the v2 regularization. However, I am getting a message that I don't know what means:

<img width="939" alt="Screenshot 2023-01-09 at 15 12 37" src="https://user-images.githubusercontent.com/117599932/211341632-8fb77cd6-0c45-4174-b992-5bcf52d379a3.png">

Can you help me understand what this means/why it is happening?

**Comment by hong-hui-zhao** [comment]
hello，guys, i meet the question same as you ,do you deal with it ,and you can tell me how to slove this question ?

**Reply by DaliBAmor** [reply]
did you find a solution plz ?

**Comment by hong-hui-zhao** [comment]
no



---Original---
From: ***@***.***&gt;
Date: Thu, May 4, 2023 01:59 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [satijalab/seurat] SCtransform: Will not return corrected UMIbecause residual type is not set to 'pearson' (Discussion #6850)




 
did you find a solution plz ?

**Comment by arjanboltjes** [comment]
It seems to me this has to do with the argument `conserve.memory = TRUE`, perhaps in combination with `vars.to.regress` not set to `NULL`. I had never used the `conserve.memory` argument until now, and I got the same error. When I remove it, the error is gone. When I use it, but without regressing out any variable, it's also gone. It's unclear to me why this would pop up.

**Comment by chloesavignac** [comment]
I have the same issue! any updates?

**Comment by yesitsjess** [comment]
I've started getting the same since adding `return.only.var.genes=F` to my `SCTransform` call

**Comment by MWRR** [comment]
same here. Is there anyone figure it out?
